src/sp/it/pl/util/async/executor/FxTimer.kt

```diff
@@ -0,0 +1,122 @@
+package sp.it.pl.util.async.executor
+
+import javafx.animation.KeyFrame
+import javafx.animation.Timeline
+import javafx.event.EventHandler
+import javafx.util.Duration
+import sp.it.pl.util.functional.invoke
+
+/**
+ * Provides factory methods for timers that are manipulated from and execute
+ * their action on the JavaFX application thread.
+ *
+ * @author Tomas Mikula (original Java code)
+ */
+class FxTimer
+/**
+ * Creates a (stopped) timer that executes the given action specified number
+ * of times with a delay period.
+ *
+ * @param delay Time to wait before each execution. The first execution is already delayed.
+ * @param cycles denotes number of executions. Use 1 for single execution, n for n executions and
+ * @param action action to execute
+ * [javafx.animation.Transition.INDEFINITE] for infinite amount.
+ */
+(delay: Duration, cycles: Int, private val action: Runnable) {
+
+    /** Equivalent to `new FxTimer(Duration.millis(delay), action, cycles);` */
+    constructor(delayMillis: Double, cycles: Int, action: Runnable) : this(Duration.millis(delayMillis), cycles, action)
+
+    constructor(delayMillis: Double, cycles: Int, action: () -> Unit) : this(Duration.millis(delayMillis), cycles, action as Runnable)
+
+    constructor(delay: Duration, cycles: Int, action: () -> Unit) : this(delay, cycles, action as Runnable)
```

Comment:
but if they are put inside the companion object, they will need to use static imports?

Reply:
Something like that, yes. The companion is for Kotlin only (since Java has problem with () -> Unit. Even then, @JvmStatic is there in moment of need. PS: I'm fairly certain you can not simply cast functional type with `as` into a Runnable, rather use `Runnable { action() }`.

I don't like this either, but it is only temporary. It is necessary to avoid getting compiler ambiguity errors when using method reference. It's unfortunate that method overloading must not be used when Kotlin's functional types are involved.